### PR TITLE
7812 Avatar double circle effect

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1355007812)

### Description of Changes Made

Implements a component called 'Avatar' which includes 2 circles that appear around the outside of the image. The circle colours are set using the css custom prop accent vars so they will update with the theme colour. 

I've created a small version as well that is used around the site.

I tried to use `inset: math.random() + px` to generate a random value so the circle appears in slightly different positions. This works but it sends the tooling into an infinite loop and I have no idea why. Updating to `sass: 1.33.0` doesn't solve it. Not sure if you have any ideas on that @helenb?

I've also created a `components.html` page that lives in the Styleguide section of the PL. I'm open to ideas on wether to proceed with this. On one hand it's good to show pattern iterations but it will require maintenance. 

### How to Test

Can be seen in the pattern library:
[http://localhost:8000/pattern-library/pattern/patterns/styleguide/components/components.html](http://localhost:8000/pattern-library/pattern/patterns/styleguide/components/components.html)

### Screenshots

<details>
  <summary>Expand to see more</summary>

Screenshot 2024-01-10 at 12 23 08

Screenshot 2024-01-10 at 12 23 27

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [x] Not required

#### Browser testing

- [x] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [x] Automated WCAG 2.1 tests pass
- [x] Manual WCAG 2.1 tests completed
- [x] I have tested in a screen reader
- [x] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] New font variants have not been added
- [ ] Not required

### CODE REVIEW

File: tbx/project_styleguide/templates/patterns/styleguide/components/components.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="p-4">
+    <p class="underline mb-2">Avatar</p>
+    {% include "patterns/atoms/avatar/avatar.html" %}
+
+    <p class="underline mt-8 mb-2">Avatar (small)</p>
+    {% include "patterns/atoms/avatar/avatar.html" with classes="avatar--small" %}
```
Comment (helenb) [inline tbx/project_styleguide/templates/patterns/styleguide/components/components.html:6]: I like this - let's see how it goes with maintaining it, and if it proves onerous we can abandon it. Could you update the design-guide docs to mention this specifically?

**Review by helenb — COMMENTED** [review]
Hi @chris-lawton thanks for this - just one small comment on the mobile styling.

I was wondering why you wanted to set a random value for the inset - does the border position vary in the designs? I can look at the tooling issue if this is important but wonder if the best thing is to go with a hard-coded value for now, and note the randomisation as a nice-to-have, if you think that it is necessary.

**Comment by chris-lawton** [comment]
> I was wondering why you wanted to set a random value for the inset - does the border position vary in the designs? I can look at the tooling issue if this is important but wonder if the best thing is to go with a hard-coded value for now, and note the randomisation as a nice-to-have, if you think that it is necessary.

I thought I saw it in different places in the design but perhaps my eyes deceived me. Let's hardcode it for now.

File: tbx/static_src/sass/components/_avatar.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        border-radius: 50%;
+        position: absolute;
+        opacity: 0.8;
+    }
+
+    &::before {
+        border: 1px solid var(--color--accent-two);
+        width: 102%;
+        height: 103%;
+        inset: -3px 2px;
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply tbx/static_src/sass/components/_avatar.scss:31]: Thanks, updated in https://github.com/torchbox/torchbox.com/pull/19/commits/4301119dbaa11f4ad227890414b3f20632c0cdb0

File: tbx/project_styleguide/templates/patterns/styleguide/components/components.html (hunk 1 above)
Comment (chris-lawton) [reply tbx/project_styleguide/templates/patterns/styleguide/components/components.html:6]: Sounds good, let's see how it goes. Aware that maintaining the pattern lib is a task in itself so conscious of not creating more work for ourselves. Added in https://github.com/torchbox/torchbox.com/pull/19/commits/01707efde56ba1a21b73f85b73009b4682ffeb3d

**Pull request merged**